src/main/java/com/google/sps/servlets/UserServlet.java

```diff
@@ -76,56 +77,68 @@
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
 
         String userID = request.getHeader("userId");
-
         if (userID == null) {
             response.sendError(500);
         } else {
             Query query = new Query("user");
             query.setFilter(new Query.FilterPredicate("userId", Query.FilterOperator.EQUAL, userID));
 
             Entity userEntity = database.prepare(query).asSingleEntity();
-            
             User user = new User(userEntity);
-            PreparedQuery results = DatastoreServiceFactory.getDatastoreService().prepare(query);
-
 
             Gson gson = new Gson();
-
             response.setContentType("application/json");
             response.getWriter().println(gson.toJson(user));
         }
     }
 
     @Override
     public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-        String jsonString = IOUtils.toString((request.getInputStream()));
-        User userInput = new Gson().fromJson(jsonString, User.class);
 
-        String userID = userInput.userId;
-        
-        // Entity with a userID as identifier.
-        Entity newUser = new Entity("user", userID);
-        newUser.setProperty("userId", userInput.userId);
-        newUser.setProperty("name", userInput.name);
-        newUser.setProperty("email", userInput.email);
-        newUser.setProperty("language", userInput.language);
-        database.put(newUser);
+        String userID = request.getHeader("userId");
+        if (userID == null) {
+            response.sendError(500);
+        } else {
+            String jsonString = IOUtils.toString((request.getInputStream()));
+            User userInput = new Gson().fromJson(jsonString, User.class);
+
+            // Entity with a userID as identifier.
+            Entity newUser = new Entity("user", userInput.userId);
```

Comment:
Define a method on user class to translate User to Entity.

Reply:
implemented